.github/workflows/unit-tests.yml

```diff
@@ -61,6 +61,58 @@ jobs:
         id: get-labels-empty
         run: echo "labels=[]" >> $GITHUB_OUTPUT
 
+
+  # This job analyzes files modified in a PR to determine which test suites should run
+  check-modified-files:
+    name: Detect Modified Tests
+    runs-on: ubuntu-latest
+    # Only run this job for pull requests (both standard PRs and copy-PR-bot branches)
+    if: github.event_name == 'pull_request' || startsWith(github.ref, 'refs/heads/pull-request/')
+    outputs:
+      # These outputs are used to conditionally trigger specific test jobs
+      run_notebook_tests: ${{ steps.check-files.outputs.run_notebook_tests }} # Set to 'true' if notebooks (.ipynb) were modified
+      run_slow_tests: ${{ steps.check-files.outputs.run_slow_tests }} # Set to 'true' if test files with pytest.mark.slow were modified
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check modified files
+        id: check-files
+        run: |
+          # Default both test flags to false - will be set to true if specific files are modified
+          RUN_NOTEBOOK_TESTS="false"  # Controls whether notebook tests will run
+          RUN_SLOW_TESTS="false"      # Controls whether slow tests will run
+
+          # Determine base and head commits for file comparison
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            # For standard PR events, GitHub provides the base and head SHAs
+            BASE_SHA=${{ github.event.pull_request.base.sha }}
```

Comment:
With copy pr bot we no longer run on pull_request targets. We could see if their action gets us this info